.github/workflows/build-test-deploy.yml

```diff
@@ -137,906 +62,417 @@ jobs:
+          PREVIEW_URL=$(vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }})
+          echo "preview-url=${PREVIEW_URL}" >> $GITHUB_OUTPUT
+        id: deploy
+      - name: Comment PR
+        uses: "./.github/templates/comment"
         with:
-          command: nix -- build .#nixfmt-check
-          token: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          message: |
+            ### Pablo Preview URL
+            ${{ steps.deploy.outputs.preview-url }}
+          comment_tag: 'Pablo Preview URL'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
 
-  deadnix-check:
-    name: "Deadnix check"
+  deploy-picasso-preview:
+    name: "Deploy Picasso Preview [FLAKY]"
     needs:
-      - check-privilege
+      - privilege-check
+      - lfs-check
+      - build-packages
     runs-on:
       - self-hosted
       - linux
       - x64
       - sre
-    concurrency:
-      group: ${{ github.workflow }}-deadnix-check-${{ github.event.pull_request.title }}
-      cancel-in-progress: true
     container:
-      image: niteo/nixpkgs-nixos-22.05:316b762afdb9e142a803f29c49a88b4a47db80ee
+      image: node:latest
     steps:
+      - run: |
+          apt-get update
+          apt-get install -y curl git
+          curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash
+          apt-get install git-lfs
       - uses: actions/checkout@v3
         with:
+          lfs: true
           ref: ${{ github.event.pull_request.head.sha }}
           persist-credentials: false
-      - uses: "./.github/templates/watch-exec"
+      - name: Install Vercel CLI
+        run: npm install --global vercel@latest
+      - name: Import project setting
+        run: |
+          mkdir -p .vercel
+          echo '${{ secrets.PICASSO_PROJ_JSON }}' >> .vercel/project.json
+      - name: Pull Vercel Environment Information
+        run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Build Project Artifacts
+        run: vercel build --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Deploy Project Artifacts to Vercel
+        run: |
+           PREVIEW_URL=$(vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }})
+           echo "preview-url=${PREVIEW_URL}" >> $GITHUB_OUTPUT
+        id: deploy
+      - name: Comment PR
+        uses: "./.github/templates/comment"
         with:
-          command: nix -- build .#deadnix-check
-          token: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          message: |
+            ### Picasso Preview URL
+            ${{ steps.deploy.outputs.preview-url }}
+          comment_tag: 'Picasso Preview URL'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
 
-  spell-check:
-    name: "Spelling check"
+  devnet-integration-tests:
+    name: Devnet integration tests [FLAKY]
     needs: 
-      - check-privilege
+      - privilege-check
+      - lfs-check
+      - build-packages
     runs-on:
       - self-hosted
       - linux
       - x64
       - sre
```

Comment:
Since the packages for integration tests will be built already, we might be able to run this on GA runners instead of our own.

Reply:
Not sure how many resources running the actual tests will take, but I'm keen to try. Applied it in 045c82b121051118fbf08af2c28c595ed2132c97